File: Day-3: Arrays Part-III/4) Majority Element(n\3).java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int n = nums.length;
        int check = n/3;
        List<Integer> ans = new ArrayList<>();
        Map<Integer, Integer> mp = new HashMap<>();
        for(int i = 0; i < n; i++)
        {
            mp.put(nums[i], mp.getOrDefault(nums[i], 0) + 1);
        }
        for(Map.Entry<Integer, Integer> it : mp.entrySet())
        {
            int count = it.getValue();
            if(count > check)
            {
                ans.add(it.getKey());
            }
        }
        return ans;
    }
}

File: Day-3: Arrays Part-III/4) Majority Element(n\3).cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int n = nums.size();
        int check = n/3;
        unordered_map<int, int> mp;
        for(int i = 0; i < n; i++)
        {
            mp[nums[i]]++;
        }
        vector<int> ans;
        for(auto i : mp)
        {
            int add = i.first;
            int count = i.second;
            if(count > check)
            {
                ans.push_back(add);
            }
        }
        return ans;
    }
};
